### PR TITLE
K8SPSMDB-1172: fix special chars in backup user/password

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPSMDB-1172](https://jira.percona.com/browse/K8SPSMDB-1172) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://perconadev.atlassian.net/browse/K8SPSMDB-1172

**DESCRIPTION**
---
**Problem:** 
_Backup user password or username with special characters cause PBM to fail._

**Cause:** 
_PBM requires a percent-encoded URI._

**Solution:**
*The operator must percent-encode usernames and passwords for the `PBM_MONGODB_URI` environment variable. These encoded usernames and passwords will be stored in an internal user secret, with each encoded value’s key appended with the `_ESCAPED` suffix.*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?
- [x] Are OpenShift compare files changed for E2E tests (`compare/*-oc.yml`)?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are all needed new/changed options added to the [Helm Chart](https://github.com/percona/percona-helm-charts)?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported MongoDB version?
- [x] Does the change support oldest and newest supported Kubernetes version?

[K8SPSMDB-1172]: https://perconadev.atlassian.net/browse/K8SPSMDB-1172?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/arbiter.log) |
| balancer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/balancer.log) |
| custom-replset-name | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/custom-replset-name.log) |
| custom-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/custom-tls.log) |
| custom-users-roles | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/custom-users-roles.log) |
| custom-users-roles-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/custom-users-roles-sharded.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/cross-site-sharded.log) |
| data-at-rest-encryption | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/data-at-rest-encryption.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/data-sharded.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/demand-backup.log) |
| demand-backup-fs | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/demand-backup-fs.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/demand-backup-eks-credentials.log) |
| demand-backup-physical | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/demand-backup-physical.log) |
| demand-backup-physical-sharded | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/demand-backup-physical-sharded.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/demand-backup-sharded.log) |
| expose-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/expose-sharded.log) |
| ignore-labels-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/ignore-labels-annotations.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/init-deploy.log) |
| finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/finalizer.log) |
| ldap | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/ldap.log) |
| ldap-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/ldap-tls.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/limits.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/liveness.log) |
| mongod-major-upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/mongod-major-upgrade.log) |
| mongod-major-upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/mongod-major-upgrade-sharded.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/monitoring-2-0.log) |
| multi-cluster-service | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/multi-cluster-service.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/non-voting.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/one-pod.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/operator-self-healing-chaos.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/pitr-sharded.log) |
| pitr-physical | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/pitr-physical.log) |
| pvc-resize | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/pvc-resize.log) |
| recover-no-primary | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/recover-no-primary.log) |
| replset-overrides | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/replset-overrides.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/rs-shard-migration.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/scaling.log) |
| scheduled-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/scheduled-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/security-context.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/self-healing-chaos.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/service-per-pod.log) |
| serviceless-external-nodes | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/serviceless-external-nodes.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/smart-update.log) |
| split-horizon | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/split-horizon.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/storage.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/tls-issue-cert-manager.log) |
| upgrade | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/upgrade.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/upgrade-consistency.log) |
| upgrade-consistency-sharded-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/upgrade-consistency-sharded-tls.log) |
| upgrade-sharded | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/upgrade-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/1a5bdf3b/version-service.log) |
| We run 53 out of 53|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1731/commits/1a5bdf3b654f2074ffca0919ce09370101ee518d
image: `perconalab/percona-server-mongodb-operator:PR-1731-1a5bdf3b`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/arbiter.log) |
| balancer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/balancer.log) |
| custom-replset-name | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/custom-replset-name.log) |
| custom-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/custom-tls.log) |
| custom-users-roles | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/custom-users-roles.log) |
| custom-users-roles-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/custom-users-roles-sharded.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/cross-site-sharded.log) |
| data-at-rest-encryption | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/data-at-rest-encryption.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/data-sharded.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/demand-backup.log) |
| demand-backup-fs | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/demand-backup-fs.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/demand-backup-eks-credentials.log) |
| demand-backup-physical | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/demand-backup-physical.log) |
| demand-backup-physical-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/demand-backup-physical-sharded.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/demand-backup-sharded.log) |
| expose-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/expose-sharded.log) |
| ignore-labels-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/ignore-labels-annotations.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/init-deploy.log) |
| finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/finalizer.log) |
| ldap | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/ldap.log) |
| ldap-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/ldap-tls.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/limits.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/liveness.log) |
| mongod-major-upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/mongod-major-upgrade.log) |
| mongod-major-upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/mongod-major-upgrade-sharded.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/monitoring-2-0.log) |
| multi-cluster-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/multi-cluster-service.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/non-voting.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/one-pod.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/operator-self-healing-chaos.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/pitr-sharded.log) |
| pitr-physical | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/pitr-physical.log) |
| pvc-resize | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/pvc-resize.log) |
| recover-no-primary | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/recover-no-primary.log) |
| replset-overrides | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/replset-overrides.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/rs-shard-migration.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/scheduled-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/security-context.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/self-healing-chaos.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/service-per-pod.log) |
| serviceless-external-nodes | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/serviceless-external-nodes.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/smart-update.log) |
| split-horizon | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/split-horizon.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/storage.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/tls-issue-cert-manager.log) |
| upgrade | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/upgrade.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/upgrade-consistency.log) |
| upgrade-consistency-sharded-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/upgrade-consistency-sharded-tls.log) |
| upgrade-sharded | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/upgrade-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1731/8b7e13cb/version-service.log) |
| We run 53 out of 53|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1731/commits/8b7e13cb01ea9896ebed88ebc9ef6febb5a91103
image: `perconalab/percona-server-mongodb-operator:PR-1731-8b7e13cb`

File: pkg/controller/perconaservermongodb/users.go [hunk 1]
```diff
@@ -134,7 +140,7 @@ func (r *ReconcilePerconaServerMongoDB) reconcileUsers(ctx context.Context, cr *
 		}
 	}
 
-	internalSysSecretObj.Data = sysUsersSecretObj.Data
+	internalSysSecretObj.Data = getInternalSecretData(&sysUsersSecretObj)
```
Comment (inelpandzic) [inline pkg/controller/perconaservermongodb/users.go:143]: Do you need `1.19.0` check here as well?

**Review by egegunes — APPROVED** [review]

File: pkg/controller/perconaservermongodb/users.go (hunk 1 above)
Comment (pooknull) [reply pkg/controller/perconaservermongodb/users.go]: https://github.com/percona/percona-server-mongodb-operator/pull/1731/commits/0a65f401d16165008dff9c32da0adf7ec9a4ff67

**Review by inelpandzic — APPROVED** [review]

**Pull request merged**